### PR TITLE
feat: add re2c

### PR DESCRIPTION
**Pull request by amaanq**
Needs https://github.com/neovim/neovim/pull/24831

### CODE REVIEW

File: queries/re2c/indents.scm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+(re2c) @indent
```
Comment (lucario387) [inline queries/re2c/indents.scm:1]: auto or smth

File: queries/re2c/injections.scm [hunk 2]
```diff
@@ -0,0 +1,5 @@
+((host_lang) @injection.content
+  (#set! injection.parent))
```
Comment (lucario387) [inline queries/re2c/injections.scm:2]: I suppose this is the section that will have `injection.parent` ?

**Comment by clason** [comment]
You could PR to main, which targets Nvim HEAD?

**Comment by amaanq** [comment]
Yeah that'd be fine, though master could benefit from re2c as well except for host_code nodes not having proper parent injections

File: queries/re2c/injections.scm [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+((define
+   option: (option_name) @_yy
+   value: (dstring) @injection.content)
+  (#any-of? @_yy "YYPEEK" "YYSKIP" "YYBACKUP" "YYBACKUPCTX" "YYRESTORE" "YYRESTORECTX" "YYFILL" "YYSHIFT")
+  (#offset! @injection.content 0 1 0 -1)
+  (#set! injection.parent))
```
Comment (lucario387) [inline queries/re2c/injections.scm:9]: missing the language or the parent here

**Comment by ObserverOfTime** [comment]
But it's broken on master.

**Comment by amaanq** [comment]
> But it's going to be broken for anyone not using HEAD.

would the query not silently fail? It's not like the query itself is invalid, just the capture

**Comment by amaanq** [comment]
the relevant PR has landed, this is good to go now

**Review by lucario387 — APPROVED** [review]

**Pull request merged**